### ISSUE TITLE
Error: cannot read properties of undefined (reading 'done') if call  if you call the function while status==='flow_resume'

### ISSUE THREAD

**Issue by AlekseyKrap** [mobxjs/mobx-state-tree#1828]
**_Bug report_**

* [x] I've checked documentation and searched for existing issues
* [x] I've made sure my project is based on the latest MST version
* [ ] Fork [this](https://codesandbox.io/s/y64pzxj01) code sandbox or another minimal reproduction. 

**Sandbox link or minimal reproduction code**
https://codesandbox.io/s/nifty-pare-7fj92?file=/index.js

**Describe the expected behavior**
I expect the flow "action" to be closed 

**Describe the observed behavior**
I am getting a global error " cannot read properties of undefined (reading 'done')"

**Comment by coolsoftwaretyler** [comment]
Hey @AlekseyKrap - I'm sorry it took so long for anyone to get back to you here. I took a look at your code and I *think* I've fixed it up in this CodeSandbox:

https://codesandbox.io/s/nice-moon-rkwmj9?file=/index.js

I know it's been a while, so perhaps you've found the answer yourself. But let me know if this is helpful and I can close it out. Or if I've missed what you really need, also let me know and we can keep troubleshooting.

If I don't hear from you in about two weeks, I'll close this out.

Thanks!
